src/main/java/org/stellar/sdk/requests/ResponseHandler.java

```diff
@@ -28,7 +29,14 @@ public T handleResponse(final Response response) throws IOException, TooManyRequ
     try {
       // Too Many Requests
       if (response.code() == 429) {
-        int retryAfter = Integer.parseInt(response.header("Retry-After"));
+        
+        Optional<Integer> retryAfter = Optional.empty();
```

Comment:
per other comment related to not using Optional for inputs, retryAfter can just be Integer here.

Reply:
Can you elaborate on that? I personally prefer to work with Optional classes everywhere, instead of nullable types. To avoid NPE by mistake somewhere

Reply:
what's the purpose of the input parameter,  either provide a value or none, can be accomplished with standard sdk, use an object(boxed primitives) and null as absent or do overloaded methods, using Optional as input parameter just adds extra layers that don't provide anything more, now you have to null check/guard on the optional first before referencing it to see if it's value is present/absent, it's duplicative. Optionals provide useful meta for return values, but not as input params.  google `java optional misuse` to see more on this line of thought.